src/main/java/SnippetManager.java

```diff
@@ -86,20 +120,78 @@ void edit(Integer id) {
         }
     }
 
-    Map<Integer, Snippet> filter(String wordToContain, String language, String[] tags) {
+    Map<Integer, Snippet> filterSnippets(String wordToContain, String language, String[] tags) {
         return snippets.entrySet()
                 .stream()
                 .filter(
                         entry -> entry.getValue().getTitle().contains(wordToContain) ||
                                 entry.getValue().getContent().contains(wordToContain)
-                )
+                ) // Match words
                 .filter(entry -> language.equals("") ||
-                        entry.getValue().getLanguage().equals(language))
+                        entry.getValue().getLanguage().equals(language)
+                ) // Match language
                 .filter(entry -> (tags.length == 1 && tags[0].equals("")) ||
-                        Arrays.asList(entry.getValue().getTags()).containsAll(Arrays.asList(tags)))
+                        Arrays.asList(entry.getValue().getTags()).containsAll(Arrays.asList(tags))
+                ) // Match tags
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
+    List<Map.Entry<Integer,Snippet>> searchSnippets(String searchTerm) {
+        ArrayList<String> terms = new ArrayList<>(Arrays.asList(searchTerm.split(" ")));
+        terms.forEach(String::toLowerCase);
+        terms.stream().map(term -> term.replaceAll("[^a-zA-Z0-9]", "")).collect(Collectors.toList());
+
+        HashMap<Integer, Integer> matchesPerSnippet = new HashMap<>();
+
+        for (String term : terms) {
```

Comment:
Maybe add comments on what each part of all the code below does, like done above. Was not immediately clear.